src/mgmt/cli_pool.c

```diff
@@ -353,3 +353,153 @@ dc_pool_destroy(tse_task_t *task)
 	tse_task_complete(task, rc);
 	return rc;
 }
+
+struct mgmt_list_pools_arg {
+	struct dc_mgmt_sys     *sys;
+	crt_rpc_t	       *rpc;
+	daos_mgmt_pool_info_t  *pools;
+	daos_size_t		req_npools;
+	daos_size_t	       *npools;
+};
+
+static int
+mgmt_list_pools_cp(tse_task_t *task, void *data)
+{
+	struct mgmt_list_pools_arg	*arg;
+	struct mgmt_list_pools_out	*pc_out;
+	struct mgmt_list_pools_in	*pc_in;
+	uint64_t			 pidx;
+	int				 rc = task->dt_result;
+
+	arg = (struct mgmt_list_pools_arg *)data;
+
+	if (rc) {
+		D_ERROR("RPC error while listing pools: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	pc_out = crt_reply_get(arg->rpc);
+	D_ASSERT(pc_out != NULL);
+	rc = pc_out->lp_rc;
+	if (rc) {
+		D_ERROR("MGMT_POOL_CREATE replied failed, rc: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	pc_in = crt_req_get(arg->rpc);
+	D_ASSERT(pc_in != NULL);
+
+	*arg->npools = pc_out->lp_npools;
+
+	/* copy RPC response pools info to client buffer, if provided */
+	if (arg->pools) {
+		/* Response ca_count expected <= client-specified npools */
+		for (pidx = 0; pidx < pc_out->lp_pools.ca_count; pidx++) {
+			struct mgmt_list_pools_one	*rpc_pool =
+					&pc_out->lp_pools.ca_arrays[pidx];
+			daos_mgmt_pool_info_t		*cli_pool =
+					&arg->pools[pidx];
+
```

Comment:
Indentation seems pretty hairy here - might be good to break out the loop internals into a helper function.